src/functions.php

```diff
@@ -42,6 +43,39 @@
 use function reset;
 use function substr;
 
+/**
+ * Check whether all servers support executing a write stage on a secondary.
+ *
+ * @internal
+ * @see https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#aggregation-pipelines-with-write-stages
+ */
+function all_servers_support_write_stage_on_secondary(Manager $manager): bool
+{
+    static $wireVersionForWriteStageOnSecondary = 13;
+
+    $servers = $manager->getServers();
+
+    /* If no servers are available, assume that SDAM has not been initialized
+     * and make a single server selection attempt to do so now. We do not
+     * actually care about selecting a server so any error can be ignored. */
+    if (empty($servers)) {
+        try {
+            $manager->selectServer(new ReadPreference(ReadPreference::NEAREST));
+        } catch (DriverRuntimeException $e) {
+        }
+
+        $servers = $manager->getServers();
+    }
+
+    foreach ($servers as $server) {
+        if (! server_supports_feature($server, $wireVersionForWriteStageOnSecondary)) {
```

Comment:
For context, this just ensures that the server's minWireVersion is <= 13 and its maxWireVersion is >= 13.

Reply:
IIUC I think this should also return true if `$server->getType() == Server::LOAD_BALANCED`. For load balanced, all backing servers are assumed to be >= 5.0.

Reply:
Can do. The current language in the CRUD spec was a compromise coming out of https://github.com/mongodb/specifications/pull/1095#discussion_r744793693. My original intention was to have the driver check the `hello` response of the backing server (which we explicitly implemented in libmongoc for the benefit of wrapping drivers); however, that was problematic because the other specs (LB, SDAM) don't currently talk about an approach for doing so. I expect some future spec is going to have to deal with that (e.g. feature requiring 5.2+).